Allow PHP 8.3

come-nc opened the pull request:
* Resolves: #38828

## Summary

The test suites in https://github.com/nextcloud/server/pull/39796 are now passing on 8.3.
This PR allows PHP 8.3 in version check, and adds it to lint CI.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required https://github.com/nextcloud/documentation/pull/11195
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

come-nc committed:
Allow PHP 8.3 in version check

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
Adding ini-file:development to 32bit CI

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
Add 8.3 to lint CI and 32bits test suite

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

nickvergessen reviewed (approved):
Dropping PHP 8.0 in a follow up?

come-nc commented:
> Dropping PHP 8.0 in a follow up?

No idea, was this decided for 28? Will it cause the same trouble as dropping 7.4? (ie there were no Nextcloud version supporting both debian stable and oldstable, making it hard to upgrade)

nickvergessen commented:
> No idea, was this decided for 28? Will it cause the same trouble as dropping 7.4? (ie there were no Nextcloud version supporting both debian stable and oldstable, making it hard to upgrade)

Basically same story applies. We have to drop it in order to upgrade to library versions that support 8.3 and well it's out of security support in November, so helping people to update is the better option.

tcitworld commented:
> Will it cause the same trouble as dropping 7.4?

8.0 was not used in major distributions according to https://repology.org/project/php/versions, except CentOS Stream 8 (where people can upgrade to 9 without issues)

come-nc committed:
Use latest shivammathur/setup-php in phpunit-32bits.yml

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

ChristophWurst reviewed (approved):
Hello

kesselb committed:
ci: clear disable_functions to enable pcntl_signal

When using ini-file development then disable_functions forbids pcntl_signal by default.

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

kesselb commented:
Pushed ef003e59565cd327d5de402d27a15f268c67e4e2 to fix `Error: Call to undefined function pcntl_signal()`

ChristophWurst commented:
> [come-nc](https://github.com/come-nc) added the[ pending documentation ](https://github.com/nextcloud/server/labels/pending%20documentation)label

add to https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.html#back-end-changes?

rinchen commented:
cron.php does not support php8.3

It reports this rather unusual statement:
`This version of Nextcloud is not compatible with PHP>=8.3.<br/>You are currently running 8.3.0.`
